sat/cli/bootprep/image.py

```diff
@@ -490,3 +491,7 @@ def create_images(instance, args):
     if args.delete_ims_jobs:
         for image in images_to_create:
             image.clean_up_image_create_job()
+
+    # TODO: This is pretty ugly, and would probably be better for it to go in
+    #  the GroupWaiter class or the future InputImageCollection class.
+    return set(waiter.members) - set(waiter.pending | waiter.failed)
```

Comment:
I tested this case, and it looks like we don't ever really get to this point of returning the images that were created if there is a failure creating images. Here's how I tested: https://gist.github.com/haasken-hpe/f15f20208da2f3dd178a9cb38dccb2d1

Basically I ran bootprep against a file that created two images, one of which depended on the other. Then I deleted the IMS image creation job for the first image to get it to fail waiting on it.

Reply:
It would probably be good to improve this like you said in the comment, but since any image creation failure causes us to raise `ImageCreateError` instead of returning output here, it's probably fine as-is for IUF. I think in the case that all images are created successfully, this set difference should be correct.